src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -456,35 +456,35 @@ public final void publishRapidResults(File jsonFile, RapidScanResultSummary summ
     //End Rapid
 
     /**
+     * Given a BDIO, creates a JSON file called {@value GenerateComponentLocationAnalysisOperation#DETECT_OUTPUT_FILE_NAME} containing
+     * every detected component's {@link ExternalId} along with its declaration location when applicable.
      * @param bdio
-     * @return JSON file containing every detected component's {@link ExternalId} along with its declaration
-     * location when applicable.
-     * @throws DetectUserFriendlyException if there was a problem generating the file.
+     * @throws OperationException
      */
-    public void generateComponentLocationAnalysisIfEnabled(BdioResult bdio) throws DetectUserFriendlyException, OperationException {
+    public void generateComponentLocationAnalysisIfEnabled(BdioResult bdio) throws OperationException {
         if (detectConfigurationFactory.isComponentLocationAnalysisEnabled()) {
-            File componentsWithLocationsFile = GenerateComponentLocationAnalysisOperation.locateComponentsForOfflineDetectorScan(bdio, directoryManager.getScanOutputDirectory());
-            publishComponentsWithLocationsFile(componentsWithLocationsFile);
+            auditLog.namedPublic(
+                    "Generate Component Location Analysis File for All Components",
+                    () -> new GenerateComponentLocationAnalysisOperation().locateComponentsForOfflineDetectorScan(bdio, directoryManager.getScanOutputDirectory(), directoryManager.getSourceDirectory())
```

Comment:
Should we be throwing an operation exception when GenerateComponentLocationAnalysisOperation.locateComponents..() fails to generate the file? I believe otherwise the status.json will include an entry like so: 
```
    {
      "startTimestamp": "2023-07-06 21:17:52.634",
      "endTimestamp": "2023-07-06 21:17:52.665",
      "descriptionKey": "Generate Component Location Analysis File for All Components",
      "status": "SUCCESS"
    }
```

Reply:
Yes. Although Detect should not fail. An info/warning message showing component location analysis could not performed for whatever reason, and status above should be failure.

Reply:
I assume a similar situation exists for generateComponentLocationAnalysisIfEnabled and that we should take the same actions.

Reply:
In order to set the status in status.json to FAILURE, we ultimately have to jump through a lot of hoops because of the way OperationRunner/Wrapper were implemented. (For example, failing to generate a Rapid Scan JSON file will still show "SUCCESS" in status.json -- can file a tech debt item to correct that and ideally implement a better way to this if others agree.). 

As for the workaround right now, I have added `ComponentLocatorException` to encompass any component locator library related exceptions/errors. Any and all exceptions that may be thrown within the library is contained and will not bubble up to Detect. The library call will return 0 or 1 based on whether everything went smoothly or if there was an issue (if enabled, debug level logging will provide sufficient info on what went wrong within the library). Please see GenerateComponentLocationAnalysisOperation.runComponentLocator(..) method and OperationWrapper.wrapped(..) for a full picture of how the custom exception is used. 

Unsuccessful Rapid scan: 
- console logs: 
![1 unsuccessful rapid scan colsole log ](https://github.com/blackducksoftware/synopsys-detect/assets/121134650/05c7ed4e-80af-42d0-a0ef-8ff62521da63)
.
.
.
![2 unsuccessful rapid console](https://github.com/blackducksoftware/synopsys-detect/assets/121134650/aa54b53d-c35e-4335-985a-84325e4595d6)
- status.json:
![unsuccess rapid status json](https://github.com/blackducksoftware/synopsys-detect/assets/121134650/4d91f179-098e-4177-9fd3-3f791625f556)
![unsuccess rapid scan status json part 2](https://github.com/blackducksoftware/synopsys-detect/assets/121134650/efe600bc-7d59-45f7-ac00-f70dba202f39)


Unsuccessful offline scan: 
- console logs: same as above
- status.json: same as above except the wording is slightly different
![offline unsuccesful status](https://github.com/blackducksoftware/synopsys-detect/assets/121134650/6a128717-ce98-4fd3-8254-bb45c20723e6)
![offline unsuccess status 2](https://github.com/blackducksoftware/synopsys-detect/assets/121134650/93744b52-59b9-483d-ba1d-85381596c22d)